app/views/response/response.html.erb

```diff
@@ -1,434 +1,553 @@
+        <% if @map.assignment.team_assignment? %>
+            <% team_member = TeamsUser.find_by_team_id(@map.contributor).user_id%>
+            <% topic_id = SignedUpTeam.topic_id(@map.assignment.id, team_member) %>
+        <% else %>
+            <% participant = Participant.find(TeamsUser.find_by_team_id(@map.contributor).user_id) %>
+            <% topic_id = SignedUpTeam.where(participant.parent_id, participant.user_id) %>
+        <% end %>
+        <% if !topic_id.nil? %>
+            <% topic = SignUpTopic.find(topic_id)%>
+            <h2>You are reviewing <%= topic.topic_name %></h2>
+            <% if !topic.description.nil? %>
+              <b>Description &nbsp;&nbsp;&nbsp;</b>
+              <% if topic.description.to_s.length>90 %>
+                <%= topic.description.to_s[0..90]+'...'%>
+                <a style="cursor: pointer" onclick="show_alert('<%= topic.description%>')">more...</a>
+              <% else %>
+                <%= topic.description.to_s%>
+              <% end %>
+              <% if topic.link != nil then %>
+                  <%= link_to image_tag('external-link-16.png', :border => 0, :align => 'middle'), topic.link, :target=>'_blank' %>
+              <% end%>
+            <% end %>
+        <% end %>
+
+        <% @submission_check = true %>
+        <%= hidden_field_tag :timeout_flag, :id => 'timeout_flag', :value => 'false' %>
+        <%= hidden_field_tag :restart_link_timing_flag, :id => 'restart_link_timing_flag', :value => 'false' %>
+        <%= hidden_field_tag :restart_file_timing_flag, :id => 'restart_file_timing_flag', :value => 'false' %>
+        <%if @assignment.teams?%>
+            <% topic_id = SignedUpTeam.topic_id(@participant.parent_id, @participant.user_id) %>
+            <%= render :partial => 'submitted_content/main', :locals => {:reviewee_participant => @contributor.participants.first,:participant => @contributor.participants.first, :stage => @assignment.get_current_stage(topic_id), :map_id => @map.id, :submission_check => @submission_check, :round => @current_round} %>
+        <%else%>
+            <% topic_id = SignedUpTeam.topic_id(@participant.parent_id, @participant.user_id) %>
+            <%= render :partial => 'submitted_content/main', :locals => {:participant => @contributor, :stage => @assignment.get_current_stage(topic_id), :map_id => @map.id, :submission_check => @submission_check, :round => @current_round} %>
+        <%end%>
+        <hr/>
+    <% end %>
+<% end %>
+
+<!-- display all versions available -->
+<% @responses_versions = Array.new %>
+<% @responses_versions = @map.get_all_versions() %>
+
+<% if @responses_versions.empty? %>
+    <I>No previous review was performed.</I><br/><hr/><br/>
+<% else %>
+    <%= render :partial => 'review', :locals => {:versions => @responses_versions} %>
+<% end %>
+<h2>Directions for the reviewer</h2>
+<input type="checkbox" id="autosave_cbx" unchecked>

```

Comment:
Since this functionality (the autosave functionality) is not actually in a fully working state and freezes in most circumstances, we have disabled this by default in our implementation. This specific change can be undone, but it makes little sense to have an enhancement feature enabled by default. This is something that the user should choose to enable rather than forcibly making them disable it every time they start a review (especially since it's broken).